lib/galaxy/tools/__init__.py

```diff
@@ -910,13 +910,13 @@ def requires_galaxy_python_environment(self):
         # seem to require Galaxy's Python.
         # FIXME: the (instantiated) tool class should emit this behavior, and not
         #        use inspection by string check
-        if self.tool_type not in ["default", "manage_data", "interactive", "data_source"]:
+        if self.tool_type not in ["default", "manage_data", "interactive", "data_source", "data_source_async"]:
             return True
 
         if self.tool_type == "manage_data" and self.profile < 18.09:
             return True
 
-        if self.tool_type == "data_source" and self.profile < 21.09:
+        if self.tool_type in ["data_source", "data_source_async"] and self.profile < 21.09:
```

Comment:
Can you split that out and make it self.profile < 24.0 ?

Reply:
Sure, if you think that's the safer option.